packages/broker/test/integration/plugins/operator/MaintainOperatorValueService.test.ts

```diff
@@ -0,0 +1,159 @@
+        logger.debug(`creating stream with streamId1 ${streamId1}`)
+        await (await streamRegistry.createStream(streamPath1, "metadata")).wait()
+        logger.debug(`creating stream with streamId2 ${streamId2}`)
+        await (await streamRegistry.createStream(streamPath2, "metadata")).wait();
+        
+        ({ operatorWallet, operatorContract } = await deployNewOperator())
+    })
+
+    it("updates both sponsorships to stay over the threshold", async () => {
+        await (await token.connect(operatorWallet).transferAndCall(operatorContract.address, parseEther("200"), operatorWallet.address)).wait()
+        // deploy 2 sponsorships, sponsor 200 & stake 100 into both of them
+        const sponsorship1 = await deploySponsorship(config, operatorWallet, { streamId: streamId1 })
+        const sponsorship2 = await deploySponsorship(config, operatorWallet, { streamId: streamId2 })
+        await (await token.connect(operatorWallet).transferAndCall(sponsorship1.address, parseEther("200"), "0x")).wait()
+        await (await token.connect(operatorWallet).transferAndCall(sponsorship2.address, parseEther("200"), "0x")).wait()
+        await (await operatorContract.stake(sponsorship1.address, parseEther("100"))).wait()
+        await (await operatorContract.stake(sponsorship2.address, parseEther("100"))).wait()
+
+        expect(await token.balanceOf(sponsorship1.address)).toEqual(parseEther("300")) // 200 sponsored + 100 staked
+        expect(await token.balanceOf(sponsorship2.address)).toEqual(parseEther("300"))
+        
+        const maintainOperatorValueService = new MaintainOperatorValueService(opertatorConfig)
+
+        const totalValueInSponsorshipsBefore = await operatorContract.totalValueInSponsorshipsWei()
+        const penaltyFraction = 0.0005 // * 1e18
+        const threshold = 200 * penaltyFraction // 0.1
+
+        // wait for sponsorships to accumulate earnings so approximate values differ enough form the real values
+        await wait(3000)
+
+        maintainOperatorValueService.start(parseEther(`${penaltyFraction}`).toBigInt()) // 200 * 0.001 = 0.1
+
+        await waitForCondition(async () => await operatorContract.totalValueInSponsorshipsWei() > totalValueInSponsorshipsBefore, 10000, 1000)
+        
+        const { sponsorshipAddresses, approxValues, realValues } = await operatorContract.getApproximatePoolValuesPerSponsorship()
+        let diff = BigInt(0)
+        for (let i = 0; i < sponsorshipAddresses.length; i++) {
+            diff = realValues[i].toBigInt() - approxValues[i].toBigInt()
+        }
+        
+        expect(await operatorContract.totalValueInSponsorshipsWei() > totalValueInSponsorshipsBefore)
+        expect(diff < threshold)
+
+        await maintainOperatorValueService.stop()
+    })
+
+    it("needs only one sponsorship to stay over the threshold", async () => {
```

Comment:
Is this test case seems to be completely identical to the previous test case? Only comments are different. Does this test something different even if there are no different code lines?

Reply:
Ups, seems like the limit fraction was overwritten somehow and wad updating only one sponsorship in both cases. Changed it back and it now updated differently for each test case.

Reply:
Will move the common setup to beforeEach and shared method